src/rcc.rs

```diff
@@ -214,91 +231,130 @@ impl CFGR {
         feature = "stm32f303xe",
         feature = "stm32f398",
     ))]
-    fn calc_pll(&self) -> (u32, u32, rcc::cfgr::PLLSRC_A) {
+    fn calc_pll(&self) -> (u32, u32, cfgr::PLLSRC_A) {
         let mut pllsrcclk = self.hse.unwrap_or(HSI / 2);
         let mut pllmul = self.sysclk.unwrap_or(pllsrcclk) / pllsrcclk;
 
         let pllsrc = if self.hse.is_some() {
-            rcc::cfgr::PLLSRC_A::HSE_DIV_PREDIV
+            cfgr::PLLSRC_A::HSE_DIV_PREDIV
         } else if pllmul > 16 {
             pllmul /= 2;
             pllsrcclk *= 2;
-            rcc::cfgr::PLLSRC_A::HSI_DIV_PREDIV
+            cfgr::PLLSRC_A::HSI_DIV_PREDIV
         } else {
-            rcc::cfgr::PLLSRC_A::HSI_DIV2
+            cfgr::PLLSRC_A::HSI_DIV2
         };
         (pllsrcclk, pllmul, pllsrc)
     }
 
     /// Returns a tuple containing the effective sysclk rate and optional pll settings.
-    fn calc_sysclk(&self) -> (u32, Option<(u8, rcc::cfgr::PLLSRC_A)>) {
+    fn calc_sysclk(&self) -> (u32, Option<(cfgr::PLLMUL_A, cfgr::PLLSRC_A)>) {
         let (pllsrcclk, pllmul, pllsrc) = self.calc_pll();
         if pllmul == 1 {
             return (pllsrcclk, None);
         }
 
         let pllmul = cmp::min(cmp::max(pllmul, 2), 16);
-        let sysclk = pllmul * pllsrcclk;
+        let sysclk = pllsrcclk * pllmul;
         assert!(sysclk <= 72_000_000);
 
-        let pllmul_bits = pllmul as u8 - 2;
-        (sysclk, Some((pllmul_bits, pllsrc)))
+        // NOTE From<u8> for PLLMUL_A is not implemented
+        // https://docs.rs/stm32f3/0.9.0/src/stm32f3/stm32f303/rcc/cfgr.rs.html#629-662
+        // let pllmul_mul: cfgr::PLLMUL_A = (pllmul as u8 - 2).into();
+        let pllmul_mul = match pllmul as u8 {
```

Comment:
I am looking to find a more elegant solution for this. After this I'm fine
merging this.

Reply:
I've found no better solution to this, as we can't implement any `From` `TryFrom` etc. trait because of the orphan rule, which is okay, because this would reduce the amount of code. 

Maybe sometime this will get implemented in `svd2rust`.